components/script/dom/window.rs

```diff
@@ -1980,6 +1980,15 @@ impl Window {
             );
         }
 
+        let document = self.Document();
+        let pending_web_fonts = self.layout.borrow().waiting_for_web_fonts_to_load();
```

Comment:
You want to scope the borrow so that it is dropped before `font_face_set.fulfill_ready_promise_if_needed();`, because since the promise calls into JS, it is very hard to be sure there will not be another borrow of `layout`.

Reply:
Isn't the lifetime of the borrow limited to the end of the statement since the `Ref` returned is a temporary and `waiting_for_web_fonts_to_load()` returns a boolean?

Reply:
Yes, sorry I forgot about the [promise enqueuing mechanism](https://github.com/servo/servo/blob/8121c9883477cf4350e8dd8abdc3776548c11ea3/components/script/script_runtime.rs#L196), so the borrow will have been dropped by the time the microtask runs.

Reply:
Actually, sorry for the confusion(unrelated to any change here since you are right about the borrow), but it appears to me that when calling `resolve_native`, the promise callback is called into then and there, not in a later microtask.

Reply:
EDIT: `resolve_native` does not go through the microtask stuff...